code/modules/surgery/advanced/advanced_neurosurgery.dm

```diff
@@ -0,0 +1,64 @@
+/datum/surgery/advanced/advanced_neurosurgery
+	name = "Advanced Neurosurgery"
+	desc = "A lengthy, precise brain surgery. Utilizes operating computer algorithms to visualize, identify and produce surgery steps to heal even the most deep-rooted of brain traumas."
+	steps = list(
+	/datum/surgery_step/incise,
+	/datum/surgery_step/retract_skin,
+	/datum/surgery_step/saw,
+	/datum/surgery_step/clamp_bleeders,
+	/datum/surgery_step/advanced_neurosurgery,
+	/datum/surgery_step/close)
+
+	target_mobtypes = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	possible_locs = list(BODY_ZONE_HEAD)
+	requires_bodypart_type = 0
+
+/datum/surgery/advanced/advanced_neurosurgery/can_start(mob/user, mob/living/carbon/target)
+	if(!..())
+		return FALSE
+	var/obj/item/organ/brain/B = target.getorganslot(ORGAN_SLOT_BRAIN)
+	if(!B)
+		return FALSE
+	return TRUE
+
+/datum/surgery_step/advanced_neurosurgery
+	name = "perform advanced_neurosurgery"
+	implements = list(
+		TOOL_HEMOSTAT = 95
+	) // No ghetto tools; this is an advanced surgery. If you have access to this, you SHOULD have access to a hemostat in the first place.
+	time = 20 SECONDS
+	preop_sound = 'sound/surgery/scalpel1.ogg'
+	success_sound = 'sound/surgery/scalpel2.ogg'
+	failure_sound = 'sound/surgery/organ2.ogg'
+	experience_given = MEDICAL_SKILL_ADVANCED //lose XP if you end up giving them bad traumas
+
+/datum/surgery_step/advanced_neurosurgery/tool_check(mob/user, obj/item/tool)
+	if(implement_type == /obj/item && !tool.get_sharpness())
```

Comment:
Dont think this proc is required since hemostats arent sharp & there arent any improvised analogues for the surgery